aptos-move/aptos-vm/src/verifier/module_init.rs

```diff
@@ -52,3 +56,49 @@ pub fn verify_module_init_function(module: &CompiledModule) -> PartialVMResult<(
     }
     Ok(())
 }
+
+/// Used for verifying an init_module function for module publishing. Used for 1.31 release and
+/// above. The checks include:
+///   1. Private visibility.
+///   2. No return types, single signer (reference) input.
+///   3. No type arguments.
+pub(crate) fn verify_init_module_function(function: &Function) -> Result<(), VMStatus> {
+    let err = |msg| Err(VMStatus::error(StatusCode::INVALID_INIT_MODULE, Some(msg)));
+
+    if !function.is_private() {
+        return err("init_module function must be private, but it is not".to_string());
+    }
+
+    if !function.return_tys().is_empty() {
+        return err(format!(
+            "init_module function must return 0 values, but returns {}",
+            function.return_tys().len()
+        ));
+    }
+
+    if function.param_tys().len() != 1 {
+        return err(format!(
+            "init_module function expects a single signer or &signer argument, \
+             but has {} arguments",
+            function.param_tys().len()
+        ));
+    }
+
+    let arg_ty = &function.param_tys()[0];
+    if !arg_ty.is_signer_or_signer_ref_ty() {
+        return err(
+            "init_module function expects a single signer or &signer argument, \
+             but its argument type is different"
+                .to_string(),
```

Comment:
Similar to above: "argument" -> "parameter".

(A function has parameters, arguments are what are passed to it at a call site)

Reply:
done